src/components/engine/configuration/system.tsx

```diff
@@ -1,21 +1,30 @@
-import { useEngineSystemHealth } from "@3rdweb-sdk/react/hooks/useEngine";
+import {
+  EngineInstance,
+  useEngineResourceMetrics,
+  useEngineSystemHealth,
+} from "@3rdweb-sdk/react/hooks/useEngine";
 import { Text } from "tw-components";
 
 interface EngineSystemProps {
-  instanceUrl: string;
+  instance: EngineInstance;
 }
 
-export const EngineSystem: React.FC<EngineSystemProps> = ({ instanceUrl }) => {
-  const { data } = useEngineSystemHealth(instanceUrl);
-  if (!data) {
+export const EngineSystem: React.FC<EngineSystemProps> = ({ instance }) => {
+  const healthQuery = useEngineSystemHealth(instance.url);
+  const metricsQuery = useEngineResourceMetrics(instance.id);
+  if (!healthQuery.data) {
```

Comment:
this means we skip rendering entirely if health query is not ready but we do not skip when rendering when metrics, is that the expected behavior?

Reply:
Yes. Metrics are a bonus. Health is the main one we care about rendering (it shows version #)